### PR TITLE
doc: update new rules for linking issue and commit message

### LINKED ISSUE #11097: Consider updating CONTRIBUTING.MD for PR issue association rule
See also https://github.com/tikv/tikv/issues/11016 & https://github.com/tikv/community/issues/149

### LINKED ISSUE #11831: updating CONTRIBUTING.MD for new commit message style
see https://github.com/tikv/community/pull/162

### PR DESCRIPTION
**Pull request by zhangyangyu**
Issue Number: close #11097, close #11831 

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by zhangyangyu** [comment]
@Mini256 @sunxiaoguang @BusyJay PTAL

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -150,45 +150,69 @@ The TiKV team actively develops and maintains a bunch of dependencies used in Ti
...
 
 ```
-<subsystem>: <what changed>
-<BLANK LINE>
-<why this change was made>
-<BLANK LINE>
-<issue reference>
-<BLANK LINE>
-Signed-off-by: <Name> <email address>
+Issue Number: close #123, ref #456
```
Comment (BusyJay) [inline CONTRIBUTING.md:172]: Is it supported by Github?

**Review by BusyJay — COMMENTED** [review]
How about also updating the PR template?

File: CONTRIBUTING.md (hunk 1 above)
Comment (zhangyangyu) [reply CONTRIBUTING.md:172]: `ref` keyword is not supported by GitHub. Multiple issues format follows GitHub's convention. It's documented in https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword.

**Comment by zhangyangyu** [comment]
> How about also updating the PR template?

Yes. After solving this PR, I'll update the PR template. It's tracked in https://github.com/tikv/community/issues/161.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -150,45 +150,69 @@ The TiKV team actively develops and maintains a bunch of dependencies used in Ti
...
+
+any multiple line commit messages that go into
+the final commit message body
+
+* fix something 1
+* fix something 2
 ```
 
-The first line is the subject and should be no longer than 50 characters, the other lines should be wrapped at 72 characters (see [this blog post](https://preslav.me/2015/02/21/what-s-with-the-50-72-rule/) for why).
+The first line is the subject(the pull request title) and should be no longer than 50 characters, the other lines should be wrapped at 72 characters (see [this blog post](https://preslav.me/2015/02/21/what-s-with-the-50-72-rule/) for why).
```
Comment (Mini256) [inline CONTRIBUTING.md:203]:
```suggestion
The first line is the subject (the pull request title) and should be no longer than 50 characters, the other lines should be wrapped at 72 characters (see [this blog post](https://preslav.me/2015/02/21/what-s-with-the-50-72-rule/) for why).
```

**Review by Mini256 — COMMENTED** [review]
LGTM

**Review by BusyJay — APPROVED** [review]

**Review by youjiali1995 — APPROVED** [review]

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -150,45 +150,69 @@ The TiKV team actively develops and maintains a bunch of dependencies used in Ti
...
+For pull requests trying to close issues in a different repository, contributors need to first create an issue in the same repository and use this issue to track.
+
+If the pull request body does not provide the required content, the bot will add the `do-not-merge/needs-linked-issue` label to the pull request to prevent it from being merged.
+
+### Format of the commit message
+
+The bot we use will extract the pull request title as the one-line subject and messages inside the `commit-message` code block as commit message body. For example, a pull request with title `pkg: what's changed in this one package` and body containing:
+
+    ```commit-message
+     any multiple line commit messages that go into
```
Comment (sunxiaoguang) [inline CONTRIBUTING.md:184]: Is the extra space in front of `any` a typo?

**Review by sunxiaoguang — APPROVED** [review]

**Pull request merged**